connector/countconnector/count.go

```diff
@@ -0,0 +1,205 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package countconnector // import "go.opentelemetry.io/collector/connector/countconnector"
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/internal/sharedcomponent"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	typeStr   = "count"
+	scopeName = "otelcol/countconnector"
+)
+
+type Config struct {
+	config.ConnectorSettings `mapstructure:",squash"`
+}
+
+var _ component.Config = (*Config)(nil)
+
+// NewFactory returns a ConnectorFactory.
+func NewFactory() component.ConnectorFactory {
+	return component.NewConnectorFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithTracesToMetricsConnector(createTracesToMetricsConnector, component.StabilityLevelDevelopment),
+		component.WithMetricsToMetricsConnector(createMetricsToMetricsConnector, component.StabilityLevelDevelopment),
+		component.WithLogsToMetricsConnector(createLogsToMetricsConnector, component.StabilityLevelDevelopment),
+	)
+}
+
+// createDefaultConfig creates the default configuration.
+func createDefaultConfig() component.Config {
+	return &Config{}
+}
+
+// createTracesToMetricsConnector creates a traces to metrics connector based on provided config.
+func createTracesToMetricsConnector(
+	_ context.Context,
+	set component.ConnectorCreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Metrics,
+) (component.TracesToMetricsConnector, error) {
+	comp := connectors.GetOrAdd(cfg.ID(), func() component.Component {
+		return newCountConnector(cfg.(*Config), set)
+	})
+
+	conn := comp.Unwrap().(*countConnector)
+	conn.metricsConsumer = nextConsumer
```

Comment:
Perhaps I'm missing something, but if you have the same connector for different input signals and different output signals, won't the last output override the previous ones? Like:

logs -> connector/1 -> metrics/prometheus1
traces -> connector/1 -> metrics/prometheus2

In the current code, won't this end up with `logs` sending data to `prometheus2`?

Reply:
In this case, each instantiation of the connector would emit to both `prometheus1` and `prometheus2` because the connector is acting as a receiver in a pipeline with two exporters. The `service` package is responsible for detecting this during build time and wrapping the exporters into a `fanoutconsumer` when appropriate.